.github/workflows/amplify_integration_tests.yaml

```diff
@@ -0,0 +1,108 @@
+name: Amplify Integration Tests
+on:
+  push:
+    branches: [main, next, stable, feat/*]
+  schedule:
+    # 6am pacific time daily, only runs on default branch
+    - cron:  '0 13 * * *'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  android:
+    runs-on: macos-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        scope: ["amplify_api_example"] #, "amplify_storage_s3_example", "amplify_auth_cognito_example"]
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+        with:
+          persist-credentials: false
+
+      # Flutter requires Java 11 to build android apps with Gradle.
+      - uses: actions/setup-java@860f60056505705214d223b91ed7a30f173f6142 # 3.3.0
+        with:
+          distribution: 'corretto' # Amazon Corretto Build of OpenJDK
+          java-version: '11'
+      
+      - name: Install dependencies
+        uses: ./.github/composite_actions/install_dependencies
+
+      - name: Fetch Amplify backend configurations
+        uses: ./.github/composite_actions/fetch_backends
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          scope: ${{ matrix.scope }}
+          api-app-id: ${{ secrets.API_APP_ID }}
+          auth-app-id: ${{ secrets.AUTH_APP_ID }}
+          datastore-app-id: ${{ secrets.DATASTORE_APP_ID }}
+          storage-app-id: ${{ secrets.STORAGE_APP_ID }}
+
+      - name: Build example app with integration tests
+        run: |
+          melos exec --scope=${{ matrix.scope }} --file-exists="integration_test/main_test.dart" --fail-fast flutter build apk --verbose
+
+      - name: Run Android integration tests
+        uses: reactivecircus/android-emulator-runner@76c2bf6f95ed6458fd659a1bdb680a0f8df232dc # 2.23.0
+        timeout-minutes: 25
+        with:
+          api-level: 29
```

Comment:
Should be 31 to match compileSdk

Reply:
Is this required? It seems there are no default images for 30 or later published so to get it working, have to change some settings which results in much slower build (adds 5-10min per build per my rough observations) https://github.com/ReactiveCircus/android-emulator-runner/issues/222

Reply:
actually beginning to think 30+ actually too slow to use after trying it more

Reply:
I agree. In general, the android emulator in this environment provided a whole slew of challenges, this just being the latest. It also seems to block building on a non-mac env bc the linux instances don't run the emulator fast enough for flutter to not timeout. Long-term, would like to spike/investigate running on real devices in device farm to avoid the emulator requirement and gain other benefits.